src/ppom-controller.ts

```diff
@@ -80,29 +71,43 @@ type PPOMVersionResponse = PPOMFileVersion[];
  * @type PPOMControllerState
  *
  * Controller state
- * @property lastChainId - ChainId for which files were last updated.
- * @property newChainId - ChainIf of currently selected network.
+ * @property chainId - ID of current chain.
+ * @property chainIdCache - Array of chainId and time it was last visited.
  * @property versionInfo - Version information fetched from CDN.
  * @property storageMetadata - Metadata of files storaged in storage.
  * @property providerRequestLimit - Number of requests in last 5 minutes that PPOM can make.
  * @property providerRequests - Array of timestamps in last 5 minutes when request was made from PPOM to provider.
  */
 export type PPOMControllerState = {
-  lastChainId: string;
-  newChainId: string;
+  // chainId of currently selected network
+  chainId: string;
+  // list of chainIds and time the network was last visited, list of all networks visited in last 1 week is maintained
+  chainIdCache: {
```

Comment:
Would a more explicit name for this be something like `chainStatus` as it's not necessarily a cache of data, but just the status of each chain.

Also, if we don't expect duplicate entries per chain ID, would it be cleaner for this to be an object keyed on the chain ID? It would speed up and simplify read operations a little and we could define a type such as `ChainStatus` for all the data such as the last visited date and whether data has been fetched.

Reply:
Good point, I updated the PR.